index.js

```diff
@@ -25,7 +27,7 @@
      * on opacity:0 or visibility:hidden.
      */
     checkComponentVisibility: function() {
-      var domnode = this.getDOMNode(),
+      var domnode = React.findDOMNode(this),
```

Comment:
it feels like this is always going to be the same `domnode` as is originally extracted in `enableVisbilityHandling`, so it might make sense to extract this during `componentDidMount` and cache it as something like `this._dom_node` instead, so we don't constantly as React to find the dom node for us. Caching it will make the mixin much faster, which is important when it comes to scrolling.